### PR TITLE
Fix the file upload for the web app

### LINKED ISSUE #374: Uploading a file on web doesn't work anymore
## Describe the bug

Since #369 has deleted the `web_firebase_file_uploader.dart` file, it's not possible to upload files on web anymore. When #221 is implemented, we can reimplement `web_firebase_file_uploader.dart`.

### PR DESCRIPTION
**Pull request by nilsreichardt**
## Description

This PR fixes the file upload for web app. The issue was #369 removed the file uploader for web because we removed the dependency of the `firebase` package.

Since the official `firebase_storage` is available for web, we can use the method from the official package. However, for web can't use `putFile` because it's not available for web. In this case, we need to use `putData`.

## Demo

Filesharing:

https://user-images.githubusercontent.com/24459435/223853461-2b24ffb9-f2e4-439d-b526-d9cfaf8d8506.mov

Homework submissions

https://user-images.githubusercontent.com/24459435/223853933-f81ab5c7-0ceb-4f5a-b59d-b1e279cff724.mov

(the cold start, lol)

## Related Tickets

Closes #374

### CODE REVIEW

File: lib/filesharing/files_basics/lib/src/local_file/local_file_data.dart [hunk 1]
```diff
@@ -34,7 +34,7 @@ class LocalFileData extends LocalFile {
 
   @override
   getFile() {
-    throw UnimplementedError();
+    return null;
```
Comment (nilsreichardt) [inline lib/filesharing/files_basics/lib/src/local_file/local_file_data.dart:37]: Returning `null` makes for me more sense (It's also used in the `_uploadToFirebase` method).

File: lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart [hunk 2]
```diff
@@ -14,14 +14,41 @@ import 'package:files_usecases/file_compression.dart';
...
+    // Defines if the file is available as Dart IO file. Normally this is the
+    // case except for files using the web app.
+    final hasIOFile = file.getFile() != null;
+
+    if (hasIOFile) {
+      // We prefer to upload the file as Dart IO file if possible.
+      //
+      // Even though the Firebase Storage plugin supports uploading files via
+      // data (Uint8List), we try to avoid this because using the Dart IO is
+      // much stable and and don't requires to load the whole file into memory
```
Comment (Jonas-Sander) [inline lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart:44]:
```suggestion
      // more stable and and doesn't require to load the whole file into memory
```

File: lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart [hunk 3]
```diff
@@ -14,14 +14,41 @@ import 'package:files_usecases/file_compression.dart';
...
+      // (can easily causing UI freezes).
```
Comment (Jonas-Sander) [inline lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart:45]:
```suggestion
      // (can easily cause UI freezes).
```

File: lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart [hunk 4]
```diff
@@ -14,14 +14,41 @@ import 'package:files_usecases/file_compression.dart';
...
+    @required fb.Reference reference,
+    @required LocalFile file,
+    fb.SettableMetadata metadata,
+  }) {
+    // Defines if the file is available as Dart IO file. Normally this is the
+    // case except for files using the web app.
+    final hasIOFile = file.getFile() != null;
+
+    if (hasIOFile) {
+      // We prefer to upload the file as Dart IO file if possible.
```
Comment (Jonas-Sander) [inline lib/filesharing/filesharing_logic/lib/src/file_uploader/implementation/firebase_file_uploader.dart:40]:
```suggestion
      // We prefer to upload the file as a Dart IO file if possible.
```

**Review by Jonas-Sander — APPROVED** [review]
LGTM, see documenation nits

**Pull request merged**